integration/integration_test.go

```diff
@@ -132,4 +135,175 @@ var _ = Describe("EiriniX", func() {
+
+			w := c.SimpleWatcherChannel(resultChan)
+
+			m.AddWatcher(w)
+
+			go m.Watch() // Start the watchers
+
+			// we shouldn't have any webhook registered
+			str, err = catalog.Kubectl([]string{}, "get", "mutatingwebhookconfiguration")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(str).To(ContainSubstring("No resources found in default namespace"))
+
+			// Generate 3 events (ADD,MODIFIED,MODIFIED)
+			app, err := c.StartEiriniApp()
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(func() bool {
+				runs, err := app.IsRunning()
+				Expect(err).ToNot(HaveOccurred())
+				return runs
+			}, time.Duration(60*time.Second), time.Duration(5*time.Second)).Should(BeTrue())
+
+			Expect(app.Delete()).ToNot(HaveOccurred()) // Delete the app (multiple events MODIFIED should be triggered)
+
+			time.Sleep(time.Second * 5) // Give time to the watcher to process the events
+
+			extension, _ := w.(*catalog.SimpleWatchChannel)
+			// Consume the recorded event
+			for _, ev := range []string{"ADDED", "MODIFIED", "MODIFIED"} {
+				event, ok := <-extension.Received
+
+				Expect(string(event.Type)).To(Equal(ev))
+				Expect(ok).To(BeTrue())
+
+				pod, ok := event.Object.(*corev1.Pod)
+				Expect(ok).To(BeTrue())
+				Expect(pod.GetName()).To(Equal(app.Name))
+			}
+
+			m.Stop()
+		})
+	})
+
+	Context("With a staging app", func() {
+		It("can see pods in the namespace", func() {
+			testns := "watchertest"
+			defer func() {
+				_, err := catalog.Kubectl([]string{}, "delete", "namespace", testns)
+				Expect(err).ToNot(HaveOccurred())
+			}()
+
+			_, err := catalog.Kubectl([]string{}, "create", "namespace", testns)
+			Expect(err).ToNot(HaveOccurred())
+
+			c := catalog.NewCatalog()
+			m := c.IntegrationManagerFiltered(false, testns)
+			resultChan := make(chan watch.Event, 3) // Test has 3 events
+
+			w := c.SimpleWatcherChannel(resultChan)
+
+			m.AddWatcher(w)
+
+			go m.Watch() // Start the watchers
+
+			// Generate 3 events (ADD,MODIFIED,MODIFIED)
+			app, err := c.StartEiriniStagingAppInNamespace(testns)
+			Expect(err).ToNot(HaveOccurred())
+			Eventually(func() bool {
+				runs, err := app.IsRunning()
+				Expect(err).ToNot(HaveOccurred())
+				return runs
+			}, time.Duration(60*time.Second), time.Duration(5*time.Second)).Should(BeTrue())
+
+			Expect(app.Delete()).ToNot(HaveOccurred()) // Delete the app (multiple events MODIFIED should be triggered)
+
+			time.Sleep(time.Second * 5) // Give time to the watcher to process the events
+
+			extension, _ := w.(*catalog.SimpleWatchChannel)
+			// Consume the recorded event
+			for _, ev := range []string{"ADDED", "MODIFIED", "MODIFIED"} {
+				event, ok := <-extension.Received
+				Expect(ok).To(BeTrue())
+
+				pod, ok := event.Object.(*corev1.Pod)
+				Expect(ok).To(BeTrue())
+				Expect(pod.GetName()).To(Equal(app.Name))
+
+				Expect(string(event.Type)).To(Equal(ev))
+				Expect(ok).To(BeTrue())
+			}
+
+			m.Stop()
+		})
+	})
+
+	Context("With a staging and an Eirini app", func() {
```

Comment:
again `"With a watcher the filters Eirini app pods"` and keep the `It` description.